### PR TITLE
feat: [PL-28659]: Infra mapping, instance sync, cd telemetry, module license entity data deletion.

### PR DESCRIPTION
**Pull request by boopesh007**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42512)

### CODE REVIEW

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 10 of 22 files at r1, 6 of 11 files at r2, 10 of 11 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42512)* status: 27 file(s) to review. 1 of 2 approvals obtained.

**Comment by boopesh007** [comment]
trigger ti0

**Comment by boopesh007** [comment]
trigger ti1

**Comment by boopesh007** [comment]
trigger ti2

**Comment by boopesh007** [comment]
trigger ti3

**Comment by boopesh007** [comment]
trigger ti4

File: 125-cd-nextgen/src/main/java/io/harness/telemetry/CdTelemetryEventListener.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
```
Comment (neelam-harness) [inline 125-cd-nextgen/src/main/java/io/harness/telemetry/CdTelemetryEventListener.java:2]: nit: should this be changed to 2023 ?

File: 126-instance/src/main/java/io/harness/service/infrastructuremapping/InfrastructureMappingServiceImpl.java [hunk 2]
```diff
@@ -62,4 +69,30 @@ public List<InfrastructureMappingDTO> getAllByInfrastructureKey(String accountId
...
+  public boolean deleteAllFromProj(String accountIdentifier, String orgIdentifier, String projectIdentifier) {
+    checkArgument(isNotEmpty(accountIdentifier), "accountId must be present");
+    checkArgument(isNotEmpty(orgIdentifier), "org identifier must be present");
+    checkArgument(isNotEmpty(projectIdentifier), "project identifier must be present");
+    try {
+      infrastructureMappingRepository.deleteAll(
+          infrastructureMappingRepository.findAllByAccountIdentifierAndOrgIdentifierAndProjectIdentifier(
+              accountIdentifier, orgIdentifier, projectIdentifier));
+    } catch (Exception e) {
+      log.error("Error while deleting infrastructure mappings present in a project {}, org {}, account{}",
```
Comment (neelam-harness) [inline 126-instance/src/main/java/io/harness/service/infrastructuremapping/InfrastructureMappingServiceImpl.java:83]: nit: missing space `account{}`

File: 930-ng-license-manager/src/main/java/io/harness/licensing/event/ModuleLicenseEventListener.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
```
Comment (neelam-harness) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/event/ModuleLicenseEventListener.java:2]: nit: change it to `2023`

**Review by neelam-harness — DISMISSED** [review]

**Pull request merged**